GauchoGrub/app/src/main/java/edu/ucsb/cs/cs190i/gauchogrub/gauchogrub/services/NotificationService.java

```diff
@@ -1,152 +1,142 @@
-                                        String addString = favorite + " - " + DiningCommon.READABLE_DINING_COMMONS[i];
-                                        // For first item of that meal type
-                                        if (!notifications.containsKey(m.event.meal.name)) {
-                                            notifications.put(mealName, new ArrayList<String>());
-                                        }
-                                        notifications.get(mealName).add(addString);
-                                    }
-                                }
+    private List<Favorite> getFavorites(String diningCommon, Calendar calendar) {
+        ArrayList<Favorite> favorites = new ArrayList<>();
+        int dayOfWeek = LocalDate.fromCalendarFields(calendar).getDayOfWeek();
+        // Get DiningCommonEntity from diningCommon string
+        DiningCommonEntity diningCommonEntityResult = data.select(DiningCommonEntity.class).where(DiningCommonEntity.NAME.eq(diningCommon)).get().first();
+        // Get RepeatedEventEntity result for current day of the week and the right dining common
+        List<RepeatedEventEntity> repeatedEvents = data.select(RepeatedEventEntity.class)
+                .where(RepeatedEventEntity.DAY_OF_WEEK.equal(dayOfWeek)
+                        .and(RepeatedEventEntity.DINING_COMMON.eq(diningCommonEntityResult))).get().toList();
+        // Get MenuEntity of the current day
+        List<MenuEntity> menusInDiningCommonToday = data.select(MenuEntity.class)
+                .where(MenuEntity.DATE.eq(LocalDate.fromCalendarFields(calendar))).get().toList();
+        // Get all favorites
+        List<FavoriteEntity> favoriteEntities = data.select(FavoriteEntity.class).get().toList();
+        for(MenuEntity menu : menusInDiningCommonToday) {
+            for(RepeatedEventEntity repeatedEvent : repeatedEvents) {
+                // If menu is today and corresponds to a repeatedEvent for the correct diningCommon
+                if(menu.getEvent().id == repeatedEvent.getId()) {
+                    // Traverse all items in the menu, add favorites to list of favorites
+                    for(FavoriteEntity favoriteEntity : favoriteEntities) {
+                        for(MenuItem menuItem : menu.getMenuItems().toList()) {
+                            // If menuItem is favorite, add to list of favorites
+                            if(menuItem.id == favoriteEntity.getId()) {
+                                favorites.add(new Favorite(menuItem, diningCommon, repeatedEvent.getMeal().name));
                             }
                         }
                     }
                 }
             }
-        } catch (Exception e) {
-            logger.info(e.getMessage());
         }
-        return notifications;
+        return favorites;
     }
 
+    private List<Favorite> getFavoritesToday(String diningCommon) {
+        return getFavorites(diningCommon, Calendar.getInstance());
```

Comment:
Let's use Joda date class to avoid conversions between Calendar and LocalDate:
`LocalDate.now()`